src/main/scala/higherkindness/rules_scala/workers/zinc/common/ColorMessage.scala

```diff
@@ -0,0 +1,28 @@
+package higherkindness.rules_scala
+package workers.zinc.common
+
+import xsbti.Problem
+import Console.{RED => CR, YELLOW => CY, GREEN => CG, RESET => CRESET}
+
+class ColorMessage() {
+  def CProblem(problem: Problem, color: String): String = {
+    val CC = color match {
+      case "RED" => CR
+      case "YELLOW" => CY
+      case "GREEN" => CG
+    }
+    problem.position.toString.size match {
+      case 0 => "[" + CC + problem.severity + CRESET + "] " + problem.message.replaceAll("\n", "\n[" + CC + problem.severity + CRESET + "] ")
+      case _ => "[" + CC + problem.severity + CRESET + "] " + problem.position + " " + problem.message.replaceAll("\n", "\n[" + CC + problem.severity + CRESET + "] ")
+    }
+  }
+
+  def CString(message: String, color: String): String = {
+    val CCSeverity = color match {
```

Comment:
Same thing as before. Additionally, the message prefix should determine color (here you have color determining prefix).

This can be broken out into methods for error, warn, and info.

Reply:
Good idea, fixed.